### PR TITLE
Toggleable AI

### PR DESCRIPTION
**Pull request by Fueredoriku**
- You can finally play with less than 4 players without AI again!
- Game can't start with only 1 player, one must either join new players or turn on AI
- Also fixes first point of #509 

TODO for another PR (will make issue on merge):
- Add helper text in player select with "Press any button to join" text for missing players

### CODE REVIEW

**Review by karofmah — APPROVED** [review]
Looks good!

File: Assets/Prefabs/UI/CreditsMenu.prefab [hunk 1]
```diff
@@ -2423,9 +2423,13 @@ PrefabInstance:
     serializedVersion: 3
     m_TransformParent: {fileID: 3901135934142223114}
     m_Modifications:
+    - target: {fileID: 1685708173701239587, guid: 1b0c0786719c84648984b75e9055e89c, type: 3}
+      propertyPath: m_text
+      value: Livia Valenti
```
Comment (toberge) [inline Assets/Prefabs/UI/CreditsMenu.prefab:2428]: Perhaps apply override to the perfab variant? Not too important unless she ends up contributing with other kinds of work somehow

File: Assets/Prefabs/UI/MenuButton.prefab [hunk 2]
```diff
@@ -86,9 +86,33 @@ MonoBehaviour:
...
+          m_ObjectArgumentAssemblyTypeName: UnityEngine.UI.Selectable, UnityEngine.UI
+          m_IntArgument: 0
+          m_FloatArgument: 0
+          m_StringArgument: 
+          m_BoolArgument: 0
+        m_CallState: 2
       - m_Target: {fileID: 0}
         m_TargetAssemblyTypeName: MainMenuMoveCamera, Assembly-CSharp
-        m_MethodName: MoveToPlayerSelect
+        m_MethodName: MoveToDefault
```
Comment (toberge) [inline Assets/Prefabs/UI/MenuButton.prefab:115]: Why did you apply this override? 🤔

File: Assets/Prefabs/UI/MenuButton.prefab [hunk 3]
```diff
@@ -237,7 +261,7 @@ MonoBehaviour:
   m_OnCullStateChanged:
     m_PersistentCalls:
       m_Calls: []
-  m_text: Label
+  m_text: Back
```
Comment (toberge) [inline Assets/Prefabs/UI/MenuButton.prefab:264]: This was supposed to be generic 😅

File: Assets/Scripts/Auction/AuctionDriver.cs [hunk 4]
```diff
@@ -81,7 +81,9 @@ private void Start()
         };
 
         playerFactory = GetComponent<PlayerFactory>();
-        playerFactory.InstantiatePlayersBidding(4 - PlayerInputManagerController.Singleton.playerInputs.Count);
+        var aiPlayerCount = PlayerInputManagerController.Singleton.MatchHasAI ?
+            4 - PlayerInputManagerController.Singleton.playerInputs.Count : 0;
```
Comment (toberge) [inline Assets/Scripts/Auction/AuctionDriver.cs:85]: Perhaps Max(x, 0) to ensure we don't end up with negative AIs sometime

File: Assets/Scripts/UI/MainMenu/MainMenuController.cs [hunk 5]
```diff
@@ -141,6 +156,35 @@ public void ReturnToMainMenu()
...
+            SelectControl(target);
+            return;
+        }
+        SelectControl(startButton);
+        if (LeanTween.isTweening(aiButtonTween))
+        {
+            LeanTween.cancel(aiButtonTween);
+            aIButton.transform.localPosition = aiButtonOriginalPosition;
+        }
+        aiButtonTween = aIButton.gameObject.LeanMoveLocal(aiButtonOriginalPosition * 1.1f, 0.3f).setEasePunch().id;
```
Comment (toberge) [inline Assets/Scripts/UI/MainMenu/MainMenuController.cs:174]: Could be pushed down to the ToggleButton class

**Review by toberge — APPROVED** [review]
Looks functional, but please left-align the text here:
image
and the wiggle animation overflows to outside the screen

**Comment by toberge** [comment]
Also, the start button isn't selected by default anymore :upside_down_face:

**Pull request merged**